api/src/test/java/com/cloud/host/ControlStateTest.java

```diff
@@ -0,0 +1,98 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.host;
+
+import com.cloud.resource.ResourceState;
+
+import junit.framework.TestCase;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ControlStateTest extends TestCase {
+
+    void verifyHostControlState(Status hostStatus, ResourceState hostResourceState, ControlState expectedControlState) {
+        Assert.assertEquals(expectedControlState, ControlState.getControlState(hostStatus, hostResourceState));
+    }
+
+    @Test
+    public void testHostControlState() {
```

Comment:
should these all be in one test? (if the first fails none of the others will be tested)